cmd/flux/check.go

```diff
@@ -172,16 +173,22 @@ func componentsCheck() bool {
 	ctx, cancel := context.WithTimeout(context.Background(), rootArgs.timeout)
 	defer cancel()
 
+	statusChecker := StatusChecker{}
+	err := statusChecker.New(time.Second, rootArgs.timeout)
+	if err != nil {
+		return false
+	}
+
 	ok := true
 	for _, deployment := range checkArgs.components {
-		kubectlArgs := []string{"-n", rootArgs.namespace, "rollout", "status", "deployment", deployment, "--timeout", rootArgs.timeout.String()}
-		if output, err := utils.ExecKubectlCommand(ctx, utils.ModeCapture, rootArgs.kubeconfig, rootArgs.kubecontext, kubectlArgs...); err != nil {
-			logger.Failuref("%s: %s", deployment, strings.TrimSuffix(output, "\n"))
+		err = statusChecker.Assess(deployment)
+		if err != nil {
+			logger.Failuref("%s: %s", deployment, err)
```

Comment:
I think we log here the controller name twice, can you please post the output of `flux check` with an error?

Reply:
Here's the output:
```bash
✗ source-controller: Status check timed out for component(s): [Deployment 'flux-system/source-controller']
✗ kustomize-controller: Status check timed out for component(s): [Deployment 'flux-system/kustomize-controller']
✗ helm-controller: Status check timed out for component(s): [Deployment 'flux-system/helm-controller']
✗ notification-controller: Status check timed out for component(s): [Deployment 'flux-system/notification-controller']
```

Reply:
Remove the component name at the beginning or the deployment id at the end from the error message?

Reply:
Instead of logging the returned error I would print something like: `source-controller: timed out waiting for rollout` and if it the check passed `source-controller: successfully rolled out`.

Reply:
In the future we could help users diagnose a failed deployment by printing the equivalent of `kubectl describe pod <controller-pod>`.

Reply:
Updated based on comment